test/cli.test.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+/* this test suit is incomplete  2015-12-18 */
+
+const test = require('tap').test;
+const request = require('request');
+const spawn = require('child_process').spawn;
+const path = require('path');
+
+const node = process.execPath;
+const defaultUrl = 'http://localhost';
+const defaultPort = 8080;
+
+function getRandomInt(min, max) {
+  return Math.floor(Math.random() * ((max - min) + 1)) + min;
+}
+
+function startEcstatic(args) {
+  return spawn(node, [require.resolve('../bin/http-server')].concat(args));
+}
+
+function checkServerIsRunning(url, msg, t, _cb) {
+  if (!msg.toString().match(/Starting up/)) {
+    return;
+  }
+  t.pass('http-server started');
+  const cb = _cb || (() => {});
+
+  request(url, (err, res) => {
+    if (!err && res.statusCode !== 500) {
+      t.pass('a successful request from the server was made');
+      cb(null, res);
+    } else {
+      t.fail(`the server could not be reached @ ${url}`);
+      cb(err);
+    }
+  });
+}
+
+function tearDown(ps, t) {
+  t.tearDown(() => {
+    ps.kill('SIGTERM');
+  });
+}
+
+const getRandomPort = (() => {
+  const usedPorts = [];
+  return () => {
+    const port = getRandomInt(1025, 65536);
+    if (usedPorts.indexOf(port) > -1) {
+      return getRandomPort();
+    }
+
+    usedPorts.push(port);
+    return port;
+  };
+})();
+
+test('setting port via cli - default port', (t) => {
+  t.plan(2);
+
+  const port = defaultPort;
+  const options = ['.'];
+  const ecstatic = startEcstatic(options);
+
+  tearDown(ecstatic, t);
+
+  ecstatic.stdout.on('data', (msg) => {
+    checkServerIsRunning(`${defaultUrl}:${port}`, msg, t);
+  });
+});
+
+test('setting port via cli - custom port', (t) => {
+  t.plan(2);
+
+  const port = getRandomPort();
+  const options = ['.', '--port', port];
+  const ecstatic = startEcstatic(options);
+
+  tearDown(ecstatic, t);
+
+  ecstatic.stdout.on('data', (msg) => {
+    checkServerIsRunning(`${defaultUrl}:${port}`, msg, t);
+  });
+});
+
+/** TODO mime types not supported in CLI
```

Comment:
I agree on commenting this out but leaving the code, but I believe the top test (using .types file) should be indirectly supported and should pass? If that's true, we can just comment out the "directly" test below.

Reply:
for mimetypes to work as CLI option (and hence in CLI test) some changes were needed. Note that the test would previously only pass because it did not assert anything related to the mimetype.